src/components/comment/AddComment.jsx

```diff
@@ -0,0 +1,52 @@
+import { useState } from "react";
+import { postComment } from "../../utils/api";
+import "./AddComment.css";
+
+function AddComment({
+	isAddingComment,
+	setIsAddingComment,
+	setComments,
+	article,
+	user,
+}) {
+	const [comment, setComment] = useState("");
+
+	const handleSubmit = e => {
+		e.preventDefault();
+		const params = {
+			body: comment,
+			author: user.username,
+		};
+		postComment(article.article_id, params).then(data => {
+			setComments(curr => [data.comment, ...curr]);
+		});
+		setIsAddingComment(false);
+	};
+
+	const cancel = () => {
+		setIsAddingComment(false);
+	};
+
+	if (!isAddingComment) return;
+
+	return (
+		<div>
+			<h3>Add Comment</h3>
+			<form onSubmit={e => handleSubmit(e)}>
+				<textarea
+					required
+					id="comment"
+					rows="4"
+					cols="50"
+					onChange={e => setComment(e.target.value)}
```

Comment:
you're just missing the value of the form, set to the comment state for this to be controlled by react.